sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -166,6 +186,23 @@ class FileStreamSource(
         // implies "sourceOptions.latestFirst = true" which we want to refresh the list per batch
         (newFiles.take(files.maxFiles()), null)
 
+      case files: ReadMaxBytes if !sourceOptions.latestFirst =>
+        // we can cache and reuse remaining fetched list of files in further batches
+        val (bFiles, usFiles) = takeFilesUntilMax(newFiles, files.maxBytes())
+        if (usFiles.map(_.size).sum < files.maxBytes() * DISCARD_UNSEEN_FILES_RATIO) {
```

Comment:
`takeFilesUntilMax` had better returns the total size together. Then,
- we can avoid this re-computation, `usFiles.map(_.size).sum`. 
- we can avoid the chance of overflow here if `takeFilesUntilMax` guarantees that once.

Reply:
I introduced `usFilesSize` that computed using `Math.addExact`. Let me know if that works. The reason I would like to leave this this way without modifying `takeFilesUntilMax` is because I want to preserve the model of `val (bFiles, usFiles) = ...` computation similar to existing `ReadMaxFiles` few lines above as it is pretty clean and simple to understand and reason about. Let me know what you think.